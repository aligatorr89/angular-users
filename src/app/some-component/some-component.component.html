<div class="" #title (click)="onClick()">
  <p>
    some-component works!
    B nubmer is: {{ b_number_test }}
  </p>
</div>
