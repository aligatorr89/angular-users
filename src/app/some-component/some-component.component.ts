import { Component, OnInit, AfterViewInit, ViewChild, ElementRef } from '@angular/core';

@Component({
  selector: 'app-some-component',
  templateUrl: './some-component.component.html',
  styleUrls: ['./some-component.component.css']
})
export class SomeComponentComponent implements OnInit, AfterViewInit {
  private a: number = 1;
  test_some_number:number = 0;
  @ViewChild('title', {static: true, read: ElementRef})
  titleSomeRef: ElementRef;
  @ViewChild('ourMain', {static: true})
  someDom: ElementRef;

  constructor() { }

  ngOnInit() {
  }

  ngAfterViewInit(): void {
  }

  onClick() {
    this.titleSomeRef.nativeElement.style.backgroundColor = 'green';
  }

}
