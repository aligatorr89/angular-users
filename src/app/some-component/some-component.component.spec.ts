import { async, ComponentFixture, TestBed, fakeAsync, flush } from '@angular/core/testing';
import { DebugElement, ElementRef } from '@angular/core';
import { By } from '@angular/platform-browser';

import { SomeComponentComponent } from './some-component.component';

describe('SomeComponentComponent', () => {
  let component: SomeComponentComponent;
  let fixture: ComponentFixture<SomeComponentComponent>;
  let someComponentDebugElement: DebugElement;
  let someComponentNativeElement: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ SomeComponentComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SomeComponentComponent);
    component = fixture.componentInstance;
    someComponentDebugElement = fixture.debugElement;
    someComponentNativeElement = fixture.nativeElement;

    fixture.detectChanges();
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('should have property test_some_number', () => {
    expect(component.test_some_number).toBeDefined();
  });

  test('should have property titleSomeRef', () => {
    expect(component.titleSomeRef).toBeTruthy();
  });

  test('should have method onClick', () => {
    expect(component.onClick).toBeDefined();
  });

  test('should have method onClick which changes background color', fakeAsync(() => {
    someComponentNativeElement.querySelector('div').click();
    expect(someComponentNativeElement.querySelector('div').style.backgroundColor).toBe('green');
    // fixture.detectChanges();
    // flush();
    // fixture.detectChanges();
  }));

  test('should have root element div', () => {
    expect(someComponentNativeElement.tagName.toLowerCase()).toBe('div');
  });

  test('should html match the required', () => {
    // expect(fixture).toMa
  });

});
