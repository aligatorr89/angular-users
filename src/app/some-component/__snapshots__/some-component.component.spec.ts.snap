// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SomeComponentComponent renders correctly 1`] = `
<app-some-component
  a={[Function Number]}
  someDom="undefined"
  test_some_number="0"
  titleSomeRef={[Function ElementRef]}
>
  <div
    class=""
  >
    <p>
       some-component works! B nubmer is:  
    </p>
  </div>
</app-some-component>
`;
